src/qibo/noise.py

```diff
@@ -158,10 +161,11 @@ class NoiseModel:
     """
 
     def __init__(self):
-        self.errors = {}
+        self.errors = collections.defaultdict(list)
+        self.conditions = collections.defaultdict(list)
```

Comment:
I think that you can do everything just by using a single collection (for example `errors`). 
See comment below.